pytensor/tensor/rewriting/linalg.py

```diff
@@ -1046,11 +1046,15 @@ def scalar_solve_to_division(fgraph, node):
     if not all(a.broadcastable[-2:]):
         return None
 
+    if core_op.b_ndim == 1:
+        # Convert b to a column matrix
+        b = b[..., None]
```

Comment:
If b had shape (5, 1) and b_ndim=1 (meaning 5 is a batch dimensions), we were returning something with the wrong shape as it became (1, 5, 1), after broadcasting with A, and then (1, 5) after squeezing below